### DISCUSSION TITLE
Tutorials at CNS*2021

### DISCUSSION THREAD

**Discussion by sanjayankur31** [OCNS/SoftwareWG#16] [Ideas]
CNS*2021 will be online from July 3 to 7: the announcement with the conference calendar is expected to be made soon. The call for tutorials page has just gone up. These are held on the first day: July 3.

https://www.cnsorg.org/cns-2021-quick

Can we, the Software WG do some tutorials? What topics do people think we can cover? I remember we'd briefly discussed hosting some beginner level sessions on common computing tools like:

- git (introduction, usage, pull request workflow, good practices)
- containers (docker/podman: using + building + sharing)

Please add your ideas here especially if you're happy to teach a tutorial. If there are other communities that do such tutorials already that we can link up with, please mention them too.

cc @OCNS/software-wg

**Comment by adolocm** [comment]
I can do a tutorial on Jupyter and Observable notebooks.
Anca

**Reply by sanjayankur31** [reply]
That sounds great! Thanks!

**Comment by sanjayankur31** [comment]
Are there other topics folks think would be good to have in a tutorial at CNS*2021?

**Reply by tnowotny** [reply]
We are planning to do a 30 min GeNN showcase for sure

**Comment by mstimberg** [comment]
A bit late to the party, but I think we will do a Brian showcase as well. If there are others interested to join in, I could also be motivated to do something general e.g. on git, general good practices (naming files, etc.) etc. It could be a half-day tutorial, or even a full day tutorial. Something mixing & matching content from the [Turing Way book](https://the-turing-way.netlify.app/welcome) for  example. Hoping to find the right balance between being interesting to many students and not too complicated to prepare :)

**Comment by stewart-heitmann** [comment]
I did  a 30-min showcase of the Brain Dynamics Toolbox last year. I liked the shorter format. I think many people want to know the general features of a software tool to help them decide whether or not it's for them. They don't necessarily care about all the nitty gritty details that you would get in a full tutorial. That said, new users who have made the switch will want more.   

Stewart Heitmann
***@***.***

**Reply by sanjayankur31** [reply]
I think everyone should at least do a showcase for their tools. That's a great way of making the scientific community aware of all the great tools out there.

From the WG's perspective, I'm wondering if we can do something tool independent as @mstimberg notes. I think we need to have a quick meeting to discuss this and divide the work + make a high level plan. We can decided on the exact sessions later, but we do need to have an idea of who is happy to take up sessions etc. to be able to submit a proposal before the deadline on May 15.

**Comment by joewgraham** [comment]
Our group is going to do a full-day tutorial on NEURON, RxD, and NetPyNE: _Building mechanistic multiscale models, from molecules to networks, using NEURON and NetPyNE_.  It should be very similar to last year's: https://www.cnsorg.org/cns-2020-tutorials#T2

I think it would be great for the WG to do a general best practices tutorial or workshop, and I would be happy to contribute.

**Comment by sanjayankur31** [comment]
Given that the tutorial day is generally booked for specific tool oriented tutorials, I was wondering if our more general tutorial could be held sometime before CNS, as "satellite tutorials" perhaps in the week or two leading up to the conference? We can discuss this with the Board and @adolocm to see if this can be done, and how. Since we'll discuss general computing skills, it may help folks pick up skills to make the specific tool oriented tutorials more useful?

Thoughts?

**Comment by clinssen** [comment]
Hi all! We are planning to do a tutorial session on modeling synaptic plasticity with NESTML, and simulating networks using the NEST Desktop graphical user interface to NEST Simulator.

Just a brainstorm idea: could we come up with a single slide, to share between all the tutorials/showcases, where we could advertise this OCNS WG and say a few words about research software engineering and its importance in neuroscience? Perhaps this could be a path towards onboarding new contributors (yes, we really do appreciate if you file a PR to fix that typo in the documentation)?

**Reply by sanjayankur31** [reply]
Sounds great! I filed a ticket for this task here:

https://github.com/OCNS/SoftwareWG/issues/22

**Comment by appukuttan-shailesh** [comment]
That is a very good idea @clinssen ! 
It will help give the Software WG more visibility.

**Comment by sanjayankur31** [comment]
@OCNS/software-wg : folks the tutorials start today in ~2 hours. Please do attend them if you can. All the links etc. are on Sched.

**Comment by sanjayankur31** [comment]
The tutorial on Git is going to start in ~5 minutes folks, zoom meeting has begun

**Comment by appukuttan-shailesh** [comment]
Tutorial on "Python for beginners" starting in around an hour (15:00 CET).
Hope to see some of you there.

**Reply by sanjayankur31** [reply]
I've started the call now @joewgraham @appukuttan-shailesh
